### PR TITLE
Apollo V3

### PR DESCRIPTION
**Pull request by devotox**
cc: @targos

* This upgrades Apollo to V3
* Changes how upload config is passed in
* Extracts `playground.ts` file from ApolloV2

We will at some point want to use the new [Apollo Landing Page](https://www.apollographql.com/docs/apollo-server/api/plugin/landing-pages/) but this is passed in a plugin and I could not figure out how to connect this properly.

For now we are able to have everything work the same way as before just using Apollo V3

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "clean": "rimraf lib",
     "eslint": "eslint src providers --ext ts --cache",
     "eslint-fix": "npm run eslint -- --fix",
-    "prepack": "npm run tsc",
+    "prepare": "npm run tsc",
```
Comment (targos) [inline package.json:36]: "prepack" is intended here

Comment (devotox) [reply package.json]: Yup. I was attempting to make this work with an install from `git` directly but it did not. I have changed it back to `prepack` and added a `postinstall`

**Comment by devotox** [comment]
@targos I am also facing this issue

```
src/schema.ts:15:17 - error TS2742: The inferred type of 'getTypeDefsAndResolvers' cannot be named without a reference to '@graphql-tools/merge/node_modules/@graphql-tools/utils'. This is likely not portable. A type annotation is necessary.

15 export function getTypeDefsAndResolvers(
                   ~~~~~~~~~~~~~~~~~~~~~~~
```

The only way i have been able to fix it is to remove declaration from `tsconfig` but I am sure that is not what you want. If you have any ideas please do tell

**Comment by targos** [comment]
@devotox thanks for the PR. I took inspiration from it and migrated to Apollo v3 in https://github.com/zakodium/adonis-apollo/commit/2b07ac0067b2f7a0ce73983125301d45b82b1020.
I installed the playground using an apollo landing page as you linked to.
I also added you as a co-author on that commit, I hope that's ok :)

**Pull request closed without merging**